Dockerfile

```diff
@@ -1,174 +1,128 @@
 # OPA BUILD STAGE -----------------------------------
-# build opa from source or download precompiled binary
+# Build OPA from source or download precompiled binary
 # ---------------------------------------------------
 FROM golang:bullseye AS opa_build
 
 COPY custom* /custom
 COPY factdb* /factdb
 
+# Build OPA binary if custom_opa.tar.gz is provided
 RUN if [ -f /custom/custom_opa.tar.gz ]; \
   then \
   cd /custom && \
   tar xzf custom_opa.tar.gz && \
   go build -ldflags="-extldflags=-static" -o /opa && \
-  rm -rf /custom ; \
+  rm -rf /custom; \
   else \
   case $(uname -m) in \
-  x86_64) \
-  curl -L -o /opa https://openpolicyagent.org/downloads/latest/opa_linux_amd64_static ; \
-  ;; \
-  aarch64) \
-  curl -L -o /opa https://openpolicyagent.org/downloads/latest/opa_linux_arm64_static ; \
-  ;; \
-  *) \
-  echo "Unknown architecture." ; \
-  exit 1 ; \
-  ;; \
-  esac ; \
+    x86_64) curl -L -o /opa https://openpolicyagent.org/downloads/latest/opa_linux_amd64_static ;; \
+    aarch64) curl -L -o /opa https://openpolicyagent.org/downloads/latest/opa_linux_arm64_static ;; \
+    *) echo "Unknown architecture." && exit 1 ;; \
+  esac; \
   fi
 
+# Build or copy factdb binary
 RUN if [ -f /factdb/factdb.tar.gz ]; \
   then \
   cd /factdb && \
   tar xzf factdb.tar.gz && \
   go build -ldflags="-extldflags=-static" -o /bin/factdb ./cmd/factstore_server && \
-  rm -rf /factdb ; \
+  rm -rf /factdb; \
   else \
   case $(uname -m) in \
-  x86_64) \
-    if [ -f /factdb/factstore_server-linux-amd64 ]; then \
-      cp /factdb/factstore_server-linux-amd64 /bin/factdb; \
-    else \
-      echo "factstore_server-linux-amd64 not found." ; \
-      if [ "$ALLOW_MISSING_FACTSTORE" = "false" ]; then \
-        echo "Missing Factstore is not allowed, exiting..."; exit 1; \
-      else \
-        echo "Missing Factstore is allowed, continuing..."; \
-        touch /bin/factdb ; \
-      fi \
-    fi \
-  ;; \
-  aarch64) \
-    if [ -f /factdb/factstore_server-linux-arm64 ]; then \
-      cp /factdb/factstore_server-linux-arm64 /bin/factdb; \
-    else \
-      echo "factstore_server-linux-arm64 not found." ; \
-      if [ "$ALLOW_MISSING_FACTSTORE" = "false" ]; then \
-        echo "Missing Factstore is not allowed, exiting..."; exit 1; \
-      else \
-        echo "Missing Factstore is allowed, continuing..."; \
-        touch /bin/factdb ; \
-      fi \
-    fi \
-  ;; \
-  *) \
-    echo "Unknown architecture." ; \
-    exit 1 ; \
-  ;; \
-  esac ; \
+    x86_64) cp /factdb/factstore_server-linux-amd64 /bin/factdb ;; \
```

Comment:
You completely removed the allow missing factstore which will break the vanilla pdp